src/components/membership-queue.ts

```diff
@@ -17,37 +18,123 @@ interface QueueUserItem {
     userId: string;
     retry: boolean;
     req: ThinRequest;
+    ts: number;
+    ttl: number;
 }
 
 export interface MembershipQueueOpts {
-    concurrentRoomLimit: number;
-    maxAttempts: number;
-    joinDelayMs: number;
-    maxJoinDelayMs: number;
+    /**
+     * The number of concurrent operations to perform.
+     */
+    concurrentRoomLimit?: number;
+    /**
+     * The number of attempts to retry an operation before it is discarded.
+     */
+    maxAttempts?: number;
+    /**
+     * @deprecated Use `actionDelayMs`
+     */
+    joinDelayMs?: number;
+    /**
+     * How long to delay a request for in milliseconds, multiplied by the number of attempts made
+     * if a request failed.
+     */
+    actionDelayMs?: number;
+    /**
+     * @deprecated Use `maxActionDelayMs`
+     */
+    maxJoinDelayMs?: number;
+    /**
+     * The maximum number of milliseconds a request may be delayed for.
+     */
+    maxActionDelayMs?: number;
+    /**
+     * How long a request can "live" for before it is discarded. This
+     * will override `maxAttempts`.
```

Comment:
This is missing a unit. With the SI base unit being seconds, I would have thought it to be seconds.
```suggestion
     * How long a request can "live" for before it is discarded in
     * milliseconds. This will override `maxAttempts`.
```